experiment_prototype/experiment_prototype.py

```diff
@@ -1171,26 +1185,75 @@ def set_slice_defaults(self, exp_slice):
 
         if 'acf' not in exp_slice:
             slice_with_defaults['acf'] = self.acf
-        if 'xcf' not in exp_slice:
             slice_with_defaults['xcf'] = self.xcf
-        if 'acfint' not in exp_slice:
             slice_with_defaults['acfint'] = self.acfint
+        elif exp_slice['acf']:
+            if 'xcf' not in exp_slice:
+                slice_with_defaults['xcf'] = True
+            if 'acfint' not in exp_slice:
+                slice_with_defaults['acfint'] = True
+        else:  # acf is False
+            # TODO log that no xcf or acfint will happen if acfs are not set.
+            slice_with_defaults['xcf'] = False
+            slice_with_defaults['acfint'] = False
+
+        if slice_with_defaults['acf']:
+            if 'rsep' in exp_slice:
+                if slice_with_defaults['rsep'] != int(round(slice_with_defaults['pulse_len'] *
+                                                            1.0e-6 * speed_of_light/2.0)):
+                    # TODO Log warning that rsep is being changed
+                    errmsg = 'Rsep was set incorrectly. Rsep will be overwritten'
+                    print(errmsg)
+                    pass
+
+            slice_with_defaults['rsep'] = int(round(slice_with_defaults['pulse_len'] * 1.0e-6 *
+                                                      speed_of_light/2.0))
+            # This is the distance travelled by the wave in the length of the pulse, divided by
+            # two because it's an echo (travels there and back).
+
+            if 'lag_table' in exp_slice:
+                # Check that lags are valid
+                for lag in exp_slice['lag_table']:
+                    if not set(np.array(lag).flatten()).issubset(set(exp_slice['pulse_sequence'])):
+                            errmsg = 'Lag {} not valid; One of the pulses does not exist in the ' \
+                                     'sequence'.format(lag)
+                            raise ExperimentException(errmsg)
+            else:
+                # build lag table from pulse_sequence
+                lag_table = list(itertools.combinations(slice_with_defaults['pulse_sequence'], 2))
+                lag_table.append([slice_with_defaults['pulse_sequence'][0], slice_with_defaults[
+                    'pulse_sequence'][0]])  # lag 0
+                lag_table.append([slice_with_defaults['pulse_sequence'][-1], slice_with_defaults[
+                    'pulse_sequence'][-1]])  # alternate lag 0
+                slice_with_defaults['lag_table'] = lag_table
+
+        else:
+            # TODO record rsep, lag_table, xcf, and acfint will not be used
+            print('Rsep, lag_table, xcf, and acfint will not be used because acf is not True.')
```

Comment:
this would be printed by experiment_handler. Both experiment_handler and radar_control have a pretty print. Not sure how I get these messages to print through that though or if they should have their own pretty print.

Reply:
can pass a function pointer to the function for pretty printing